lib/spdy-transport/connection.js

```diff
@@ -281,7 +281,7 @@ Connection.prototype._handleFrame = function _handleFrame(frame) {
 
   // Session window update
   if (frame.type === 'WINDOW_UPDATE' && frame.id === 0) {
-    if (state.version < 3.1 && state.autoSpdy31) {
```

Comment:
I believe the logic here was: if we receive `id=0 WINDOW_UPDATE` and we are on a version `3` (or less, yeah it should be `=== 3` here) and we want to auto-upgrade - do it.

Why do you suggest that `autoSpdy31` should be removed from this `if`?